### ISSUE TITLE
Support for more "exotic" window types

### ISSUE THREAD

**Issue by EiniStein** [LorenzCK/OnTopReplica#15]
Hello,

I am using OTR to force some windows on top that cannot be made on top with any other application. 

One such window is Windows' own Resource Manager that OTR is able to display, but unfortunately it does not allow click-forwarding. Maybe this could be solved, so that the original window could be minimized out of the way?

There are seem to be window types that OTR does not catch at all. One such example is the sensors window of HWInfo. This is another program that does not come with its own "always on top" function, so OTR could help out once it is able to detect HWInfo's window.

Thanks and regards.

**Comment by LorenzCK** [comment]
Click-forwarding is difficult to do, without letting the input focus switch to the target window. It largely depends on how the input handling of the target windows is implemented. I will look into the Windows Resource Manager.

About the windows filtering: this part of OTR needs some fixes anyhow because of how Windows Store applications are shown in the menu. By giving a look to [HWInfo](https://www.hwinfo.com/), it appears that many windows are shown as "tool" windows, which are usually filtered out by OTR because they are not meant to show information. It might be a good idea to add an option to put "tool" windows in the selection. 👍

**Comment by EiniStein** [comment]
Once click-forwarding works for Resource Manager it could be useful to have an option like shift-click on the border that allows to move the window instead of resizing it. That way it could stay in click-through mode and still be moved around.

An option to automatically minimize the target window could also be useful, for those of use who only need the replica anyway. Even more so when click-forwarding can be use.

PS: OTR's own "Minimize" behaves odd here. Once it is minimized I have to click *twice* on its task bar symbol to restore it. Also the help mentions minimizing to the tray, how is this done?

**Comment by EiniStein** [comment]
I just learned how click-forwarding can be made to work with Resource Manager. OTR needs administrator rights, likely because Resource Manager runs using administrator rights.

Unfortunately it turns out that RM stops refreshing its display once its minimized. So both windows need to stay on the screen. This brings me to another feature idea: a position lock that keeps the OTR window on top of its target window and makes it move with the target window (preferably with border turned off).

**Comment by LorenzCK** [comment]
> Once click-forwarding works for Resource Manager it could be useful to have an option like shift-click on the border that allows to move the window instead of resizing it. That way it could stay in click-through mode and still be moved around.

Makes sense. Opened issue #17 for this.

> I just learned how click-forwarding can be made to work with Resource Manager. OTR needs administrator rights, likely because Resource Manager runs using administrator rights.

Didn't think about that one. Glad it works! 😊

> Unfortunately it turns out that RM stops refreshing its display once its minimized.

Yes, this is the default behavior of Windows's windowing manager. Minimized windows do not refresh their contents and their thumbnails stay "frozen".

> a position lock that keeps the OTR window on top of its target window and makes it move with the target window.

Could you explain a use-case for this scenario? I can't see how having an exact replica of a target window on top of the target window itself can be useful. The usual use-case (as far as I know) is to have a replica of a window that is not visible (because it is occluded by other windows), therefore having it move with OTR doesn't seem to make much sense to me (i.e., who cares _where_ the original window is, as long as the replica is visible). I'm probably missing something. 😄

> Also the help mentions minimizing to the tray, how is this done?

This was not updated. OTR minimize behavior depends on the Windows version: on Windows 7 and greater (where the task bar exists), it minimizes to the task bar only. On Vista it minimizes to tray.
Some options on this could/should probably be added in the future.

**Comment by EiniStein** [comment]
I thought to abuse OTR to force some windows always on top that other software failed to do. Now I noticed that the culprit was the same administrator rights that kept OTR from doing its job.

So forget about this feature request, I will just use 4t Tray Minimizer for those occasions where I just need a window on top without wanting a (smaller) duplicate around.

**Comment by EiniStein** [comment]
Yes, this is a function I use regularly. But since other software does provide this already I do not know if it is worth your effort. The main advantage of the close seems to be that it can be resized. When(ever) this is not needed than having the original window on top seems more useful.
